openevsewifi/__init__.py

```diff
@@ -27,22 +28,65 @@
 
 colors = ['off', 'red', 'green', 'yellow', 'blue', 'violet', 'teal', 'white']
 
+def parse_checksum(s):
+    """
+    If there is a '^' in given string s, this checks that the xor of utf8 bytes
+    before the '^' equal the hex value specified after '^'.  It returns the
+    string before the '^' on success, None on error.
+
+    If there is no '^' in the string, the string is returned.
+    """
+    spl = s.rsplit('^', 1)
+    if len(spl) == 1:
+        return s
+    try:
+        check = int(spl[1], 16)
+    except:
+        return None
+    datsum = 0
+    for c in spl[0].encode('utf-8'):
+        datsum ^= c
+    if datsum != check:
+        return None
+    return spl[0]
+
+def json_parser(s):
+    print("json: ", s)
+    try:
+        result = json.loads(s)
+        print("result: ", result)
+        parsed = parse_checksum(result["ret"])
+        if parsed is not None:
+            parsed = parsed[1:] # strip of leading $
+            parsed = parsed.split()
+        return parsed
+    except Exception as e:
```

Comment:
Bare exceptions like this are generally a bad idea.  What exceptions are you trying to account for here?  The library _should_ handle better handle errors in a way that lets the consuming application determine the best course of action (I'll open an issue for that), but returning None here is problematic (see my [comment](https://github.com/miniconfig/python-openevse-wifi/pull/7/files#r450971993) below)

Reply:
Fixed.